files/build_templates/sonic_debian_extension.j2

```diff
@@ -684,8 +691,20 @@ sudo LANG=C DOCKER_HOST="$DOCKER_HOST" chroot $FILESYSTEM_ROOT docker tag {{imag
 sudo LANG=C DOCKER_HOST="$DOCKER_HOST" chroot $FILESYSTEM_ROOT docker tag {{imagename}}:latest {{imagebasename}}:latest
 {% endif %}
 fi
+}
+DLIST+=({{imagename}})
+export -f {{imagename}}
 {% endfor %}
 
+#### RUN in parallel if RAMDISK is enabled  ####
+if [ ! -z "${USE_TMPFS}" ]; then
+	NPROC=$(nproc)
+	if [ ${NPROC} -gt 16 ]; then NPROC=16; fi
+	printf '%s\n' "${DLIST[@]}" | xargs -I{} -P ${NPROC}  bash -c " SONIC_IMAGE_VERSION=${SONIC_IMAGE_VERSION} {}"
+else
+	printf '%s\n' "${DLIST[@]}" | xargs -I{} -P 1  bash -c "SONIC_IMAGE_VERSION=${SONIC_IMAGE_VERSION} {}"
+fi
+
```

Comment:
Could you please improve the logic? The line 703 and 705 can be called only once.
```
NPROC=$(nproc)
[ $NPROC > 16 ]  && NPROC=16
[ In some conditions ] && NPROC=1
printf '%s\n' "${DLIST[@]}" | xargs -I{} -P ${NPROC}  bash -c " SONIC_IMAGE_VERSION=${SONIC_IMAGE_VERSION} {}"
```